### PR TITLE
Simplify CellExplorer sorting with pymat reader

### PR DESCRIPTION
**Pull request by h-mayorquin**
When I was working in a conversion I started used this reader for matlab files. Pymat reader uses a consistent interface for matlab files that are in the old format and in hdf5. This allows us to simplify the if conditions code that we have in `CellExplorerSortingExtractor` and have a cleaner code.

Plus, I added some conditions to avoid reading waveform data when possible and a third condition for extracting sampling frequency from the session file when this is available.

### CODE REVIEW

File: src/spikeinterface/extractors/cellexplorersortingextractor.py [hunk 1]
```diff
@@ -149,15 +150,29 @@ def __init__(
         sorting_segment = CellExplorerSortingSegment(spiketrains_dict, unit_ids)
         self.add_sorting_segment(sorting_segment)
 
-        self.extra_requirements.append("scipy")
+        self.extra_requirements.append(["scipy", "pymatreader"])
```
Comment (alejoe91) [inline src/spikeinterface/extractors/cellexplorersortingextractor.py:153]: Do we still need `scipy` then?

Comment (h-mayorquin) [reply src/spikeinterface/extractors/cellexplorersortingextractor.py:153]: This other library usew scipy under the hood, where do we use this extra requirements btw?

Comment (alejoe91) [reply src/spikeinterface/extractors/cellexplorersortingextractor.py:153]: In the containers, to install all requirements to reload the extractor ;) If `scipy` is a requirement of `pymatreader`, then we can drop it!

Comment (h-mayorquin) [reply src/spikeinterface/extractors/cellexplorersortingextractor.py:153]: Thanks, let's drop it.

Comment (h-mayorquin) [reply src/spikeinterface/extractors/cellexplorersortingextractor.py]: Removed.

**Review by alejoe91 — APPROVED** [review]

**Pull request merged**